agent/testflinger_agent/client.py

```diff
@@ -365,3 +365,39 @@ def post_influx(self, phase, result=None):
             )
         except InfluxDBClientError as exc:
             logger.error(exc)
+
+    def post_status_update(self, job_queue, webhook, events):
+        """
+        Posts status updates about the running job as long as there is a
+        webhook
+
+        :param job_queue:
+            TestFlinger queue the currently running job belongs to
+        :param webhook:
+            String URL to post status update to
+        :param events:
+            List of accumulated test events
+
+        """
+        if webhook is None:
+            return
```

Comment:
I think this is actually ok. While it's true that we DO want to expose some information about recent provision failures on the testflinger UI under the agent detail page, I think it will have to work a little different. Instead of discrete events like this, we are really only concerned with why the provision phase finished, and probably storing it in a way that is easier to retrieve when we retrieve the other agent data. I'm going to be looking into that piece soon, and if it turns out that we can reuse this then I'll certainly try to do so. But the call for sending stuff along to test-observer is already a little strange and probably best to keep it separate for now.

Reply:
It does feel strange though that the agent would know about the webhook. wdyt?